jog_arm/src/jog_arm/jog_calcs.cpp

```diff
@@ -614,20 +614,23 @@ namespace jog_arm {
 
     // Store joints in a member variable
     for (std::size_t m = 0; m < command.joint_names.size(); ++m) {
-      for (std::size_t c = 0; c < jt_state_.name.size(); ++c) {
-        if (command.joint_names[m] == jt_state_.name[c]) {
-          // Apply user-defined scaling if inputs are unitless [-1:1]
-          if (parameters_.command_in_type == "unitless")
-            result[c] = command.deltas[m] * parameters_.joint_scale;
-            // Otherwise, commands are in m/s and rad/s
-          else if (parameters_.command_in_type == "speed_units")
-            result[c] = command.deltas[m] * parameters_.publish_period;
-          else
-            ROS_ERROR_STREAM_NAMED(NODE_NAME, "Unexpected command_in_type");
-          goto NEXT_JOINT;
-        }
+      std::size_t c;
+      try {
+        c = jt_state_name_map_.at(incoming_jts_.name[m]);
```

Comment:
I guess there must be an internal loop through jt_state_name_map_ here, so there isn't really an improvement from O(n^2) to O(n), right?

Reply:
`jt_state_name_map_` is a `std::map` (ie: a hash map). Those have O(n) cost for lookups. There is no loop.

Reply:
OK, cool. But it's inside a loop:

> for (std::size_t m = 0; m < command.joint_names.size(); ++m)

Anyway, I'll keep this change.

Reply:
`N * O(log N) <= O(N^2)`.

Reply:
Thanks!